### PR TITLE
Cleanup: ensure no service account is used unless required

### PR DESCRIPTION
**Pull request by dduportal**
Kubernetes Service Accounts are only required to interact with a Kubernetes API.

The helm charts changed in this PR has no need to interact with the Kubernetes API, so all references to service account object is removed and all pod templates are explicitly disabling the SVC account automounting (no more token to authenticate to the Kubernetes API).

All references to the service account are removed from the helm charts:
- template functions
- docs
- SVC Account templates, 
- pods attribute serviceaccount
- values

💡 Please note:

- There is 1 commit per chart change. We must use a "Rebase and Merge" when mergin the PR to keep the commit history (no squash, no merge!)
- Some helm charts only had template helper which result in no change (patch bump only)
- Some helm charts only have the service account automounting disabled

### CODE REVIEW

File: charts/artifact-caching-proxy/Chart.yaml [hunk 1]
```diff
@@ -2,4 +2,4 @@ apiVersion: v2
 description: artifact-caching-proxy is a Nginx caching proxy in front of repo.jenkins-ci.org
 name: artifact-caching-proxy
 type: application
-version: 0.11.1
+version: 0.11.2
```
[the remark this answers is not in the document]
Comment (dduportal) [reply charts/artifact-caching-proxy/Chart.yaml:5]: true that, I did the cleanup in 2 times (remove svc account remnants, then disable automount) and it made me mess up the version. thanks!

File: charts/basic-helm-chart-example/Chart.yaml [hunk 2]
```diff
@@ -15,10 +15,10 @@ type: application
...
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.1.1
+version: 0.2.0
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
 # It is recommended to use it with quotes.
-appVersion: "1.16.0"
+# appVersion: "v1.0.0"
```
[the remark this answers is not in the document]
Comment (dduportal) [reply charts/basic-helm-chart-example/Chart.yaml:24]: I did not realize it was commented out, I've most probably commited something unwanted. Good catch

[the remark this answers is not in the document]
Comment (dduportal) [reply charts/basic-helm-chart-example/Chart.yaml:24]:
```suggestion
appVersion: "1.16.0"
```

**Review by lemeurherve — APPROVED** [review]
👏

**Pull request merged**